csm/core/agent/api.py

```diff
@@ -249,31 +249,63 @@ def getMatchingEndpoint(endpoint_map, path):
         else:
             Log.debug(f"Feature endpoint is not found for {request.path}")
 
+    @classmethod
+    def _extract_bearer(cls, headers) -> str:
+        """
+        Extract the bearer token from HTTP headers.
+
+        :param headers: HTTP headers.
+        :returns: bearer token.
+        """
+
+        hdr = headers.get(CsmAuth.HDR)
+        if not hdr:
+            raise CsmNotFoundError(f'No {CsmAuth.HDR} header')
+        auth_pair = hdr.split(' ')
+        if len(auth_pair) != 2:
+            raise CsmNotFoundError(f'The header is incorrect. Expected "{CsmAuth.HDR} session_id"')
+        auth_type, session_id = auth_pair
```

Comment:
Does it make sense to use something like
```
auth_type, session_id = hdr.split(' ', 2)
```
Instead, and have the `_validate_bearer()` method take care of broken session IDs instead?  Please consider any potential risks, this is only a suggestion of code change.

Reply:
I'd rather consider these checks as a part of successful extraction. Let's keep it for now, if you don't mind.